release-notes/latest.md

```diff
@@ -19,6 +19,10 @@
 
 * All API payloads are now gzipped. This should help save bandwidth and make the web UI load faster. #4470
 
+#### <sub><sup><a name="4494" href="#4494">:link:</a></sup></sub> feature
+
+* Optimize certain API requests to use single transaction of DB, instead of making DB connection for each sub-query. #4494
```

Comment:
```suggestion
* When an API request is sent from the Web UI and there is no response, it sends another request every 5 seconds. This can clobber the database with a ton of the same requests and cause the first request to be delayed further because of the competition to grab a connection. This PR optimizes all API requests to use single transaction talking to the database, which will result in the queries for each request to be completed together.  #4494
```

Reply:
@clarafu Nothing wrong with adding too much info in release notes; I'd rather over-communicate than under-communicate as contributors will be reading these, too.

Suggesting a different wording though:

```suggestion
* API endpoints have been changed to use a single transaction per request, so that they become "all or nothing" instead of holding data in memory while waiting for another connection from the pool. This could lead to snowballing and increased memory usage as requests from the web UI (polling every 5 seconds) piled up. #4494
```